### PR TITLE
fix: rescheduling daily

### LINKED ISSUE #13563: [CAL-3084] self-rescheduled meeting didn't have cal.com meeting
Found a bug? Please fill out the sections below. 👍

### Issue Summary

image

I rescheduled the meeting myself (twice) and the [cal.com](http://cal.com) meeting got lost ( see above )

This was the meeting: [https://app.cal.com/video/8Pyt9CDagAw8rghxtQbvvG](https://app.cal.com/video/8Pyt9CDagAw8rghxtQbvvG)

### Steps to Reproduce

1. (for example) Went to ...
2. Clicked on...
3. ...

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Actual Results

* What's happening right now that is different from what is expected

### Expected Results

* This is an ideal result that the system should get after the tests are performed

### Technical details

* Browser version, screen recording, console logs, network requests: You can make a recording with [Bird Eats Bug](https://birdeatsbug.com/).
* Node.js version
* Anything else that you think could be an issue.

### Evidence

* How was this tested? This is quite mandatory in terms of bugs. Providing evidence of your testing with screenshots or/and videos is an amazing way to prove the bug and a troubleshooting chance to find the solution.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3084](https://linear.app/calcom/issue/CAL-3084/self-rescheduled-meeting-didnt-have-calcom-meeting)</sub>

### LINKED ISSUE #14502: [CAL-3389] Rescheduling bug
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Reported on x [https://twitter.com/jonathankerrr/status/1773012340071428166](https://twitter.com/jonathankerrr/status/1773012340071428166)

image

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3389](https://linear.app/calcom/issue/CAL-3389/rescheduling-bug)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/13563 
Fixes https://github.com/calcom/cal.com/issues/14502
Fixes CAL-3389
Rescheduling works fine for daily video but doesn't work when your location is empty string. It could be possible when there is no location selected on event type settings.
Screenshot 2024-02-17 at 1 11 30 AM

updateMeeting(in dailyvideo/lib/VideoApiAdapter) was never called because location is empty string.

How to reproduce the bug?
1) Remove all location from event type settings
2) Book a meeting
3) Reschedule the meeting

Then check the location url is daily video url and check sidebar if it is updated to new date and time 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by sean-brydon — DISMISSED** [review]
Tested locally works fine

**Review by CarinaWolli — COMMENTED** [review]
Webhook e2e tests are failing because of the changes

**Review by CarinaWolli — DISMISSED** [review]

**Comment by Udit-takkar** [comment]
Idk tests are working fine locally 

Screenshot 2024-02-21 at 8 12 40 PM

**Comment by keithwillcode** [comment]
> > @Udit-takkar Why weren't tests added to confirm this change?
> 
> I already added one test for this #13103 some time ago (when location is daily video). only difference is when location is empty string update it to "integrations:daily" which is verified by webhook test

Ok. Ideally moving forward we write more direct tests ensuring logic is correct. The webhook test is good but a bit indirect verification of the logic versus direct verification of the logic.

**Comment by CarinaWolli** [comment]
@Udit-takkar any idea why tests are failing here?

**Comment by Udit-takkar** [comment]
> @Udit-takkar any idea why tests are failing here?

I have fixed one but don't know why this test is failing. It is passing successfully locally.

Screenshot 2024-02-28 at 12 05 58 PM

**Comment by CarinaWolli** [comment]
> I have fixed one but don't know why this test is failing. It is passing successfully locally.

`DAILY_API_KEY` isn't set and that's why createMeeting fails and emails aren't sent. @keithwillcode can we set `DAILY_API_KEY` for e2e tests?

**Comment by keithwillcode** [comment]
> DAILY_API_KEY

It's already set up as `CI_DAILY_API_KEY`. Just need to map it correctly in the workflow actions files.
https://github.com/calcom/cal.com/blob/7422488b5dab650259efde45e0b64763dd432757/.github/workflows/e2e.yml#L47
https://github.com/calcom/cal.com/blob/7422488b5dab650259efde45e0b64763dd432757/.github/workflows/e2e-embed.yml#L49
https://github.com/calcom/cal.com/blob/7422488b5dab650259efde45e0b64763dd432757/.github/workflows/e2e-app-store.yml#L56
https://github.com/calcom/cal.com/blob/7422488b5dab650259efde45e0b64763dd432757/.github/workflows/e2e-embed-react.yml#L43

File: .github/workflows/e2e-embed.yml [hunk 1]
```diff
@@ -74,6 +74,7 @@ jobs:
           SENDGRID_EMAIL: ${{ secrets.CI_SENDGRID_EMAIL }}
           TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
           TURBO_TEAM: ${{ secrets.TURBO_TEAM }}
+          DAILY_API_KEY: ${{ secrets.CI_DAILY_API_KEY }}
```
Comment (keithwillcode) [inline .github/workflows/e2e-embed.yml:77]: These were put in alphabetical order on our oírse in all of these files. Let’s maintain that

**Review by keithwillcode — DISMISSED** [review]

File: apps/web/playwright/booking-pages.e2e.ts [hunk 2]
```diff
@@ -80,12 +80,6 @@ testBothFutureAndLegacyRoutes.describe("free user", () => {
     await expect(page.locator("[data-testid=success-page]")).toBeVisible();
     const { title: eventTitle } = await user.getFirstEventAsOwner();
 
-    await expectEmailsToHaveSubject({
-      emails,
-      organizer: user,
-      booker: bookerObj,
-      eventTitle,
-    });
```
Comment (Udit-takkar) [inline apps/web/playwright/booking-pages.e2e.ts:88]: this works fine locally but fails on CI.

Comment (Udit-takkar) [reply apps/web/playwright/booking-pages.e2e.ts:88]: added a new unit test "cannot book same slot multiple times" in fresh-booking to make sure which would check the email has been delivered to both booker and organizer

**Comment by CarinaWolli** [comment]
> 1. Remove all location from event type settings
> 2. Book a meeting
> 3. Reschedule the meeting
> Then check the location url is daily video url and check sidebar if it is updated to new date and time

I tried that in production and the daily link did show the updated date and time. How do I reproduce this bug?

**Review by CarinaWolli — APPROVED** [review]
Looks good 🙏

**Pull request merged**